File: spring-user/src/main/java/com/example/springuser/controller/GlobalExceptionHandler.java
package com.example.springuser.controller;

import com.example.springuser.dto.BaseResponse;
import com.example.springuser.dto.ResultCode;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;

@ControllerAdvice
public class GlobalExceptionHandler {

    @ExceptionHandler(IllegalArgumentException.class)
    public ResponseEntity<BaseResponse<String>> handleIllegalArgumentException(IllegalArgumentException e) {
        BaseResponse<String> response = new BaseResponse<>();
        response.setResultCode(ResultCode.ERROR.name());
        response.setMessage(e.getMessage());
        return ResponseEntity.badRequest().body(response);
    }

    // 다른 예외들에 대한 처리가 필요할 경우 추가
}

File: spring-kotlin-sync/src/main/kotlin/com/example/kotlin/presentation/exception/GlobalExceptionHandler.kt
package com.example.kotlin.presentation.exception

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.http.converter.HttpMessageNotReadableException
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.context.request.WebRequest

@ControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(Exception::class)
    fun handleAllExceptions(ex: Exception, request: WebRequest): ResponseEntity<com.example.kotlin.presentation.exception.ErrorResponse> {
        val errorResponse = ErrorResponse(
            statusCode = HttpStatus.INTERNAL_SERVER_ERROR.value(),
            message = ex.localizedMessage
        )
        return ResponseEntity(errorResponse, HttpStatus.INTERNAL_SERVER_ERROR)
    }

    @ExceptionHandler(HttpMessageNotReadableException::class)
    fun handleHttpMessageNotReadableException(ex: HttpMessageNotReadableException, request: WebRequest): ResponseEntity<com.example.kotlin.presentation.exception.ErrorResponse> {
        val errorResponse = ErrorResponse(
            statusCode = HttpStatus.BAD_REQUEST.value(),
            message = "Invalid input"
        )
        return ResponseEntity(errorResponse, HttpStatus.BAD_REQUEST)
    }

    @ExceptionHandler(NoSuchElementException::class)
    fun handleNoSuchElementException(ex: NoSuchElementException, request: WebRequest): ResponseEntity<com.example.kotlin.presentation.exception.ErrorResponse> {
        val errorResponse = ErrorResponse(
            statusCode = HttpStatus.NOT_FOUND.value(),
            message = ex.localizedMessage
        )
        return ResponseEntity(errorResponse, HttpStatus.NOT_FOUND)
    }

    // 필요에 따라 다른 예외 처리기 추가
}